packages/web/app/api/TamanuApi.jsx

```diff
@@ -135,6 +131,11 @@ export class TamanuApi extends ApiClient {
     });
   }
 
+  setToken(token, refreshToken) {
+    localStorage.setItem(TOKEN, refreshToken);
+    return super.setToken(token, refreshToken);
+  }
```

Comment:
by overriding this method we ensure we always save the token regardless of when this is called, saving us from separately handling all the different situations